### PR TITLE
Bug fix for #1662

### LINKED ISSUE #1662: Simulator server cannot start in current dev branch
### Versions

- Python: 3.9
- OS: macos 13.4
- Pymodbus: dev-3.4.0
- Modbus Hardware (if used): None

### Pymodbus Specific

- Server: tcp, simulator
- Client: None

### Description

When I was trying to start a simulator by running main.py in server/simulator/, python complained with error logs below:

```bash
pymodbus $ PYTHONPATH=. python server/simulator/main.py
2023-07-18 09:23:23,159 INFO  logging:96 Start simulator
Executing <Task finished name='Task-1' coro=<run_main() done, defined at /Users/hydra/Documents/coding/pymodbus/pymodbus/server/simulator/main.py:112> exception=TypeError("__init__() got an unexpected keyword argument 'loop'") created at /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py:621> took 0.118 seconds
Traceback (most recent call last):
  File "xxx/pymodbus/server/simulator/main.py", line 125, in <module>
    main()
  File "xxx/pymodbus/server/simulator/main.py", line 121, in main
    asyncio.run(run_main(), debug=True)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "xxx/pymodbus/server/simulator/main.py", line 115, in run_main
    task = ModbusSimulatorServer(**cmd_args)
  File "xxx/pymodbus/server/simulator/http_server.py", line 168, in __init__
    self.modbus_server = comm(framer=framer, context=datastore, **server)
TypeError: __init__() got an unexpected keyword argument 'loop'
```

I've digged into the commit logs of pymodbus and found that class ModbusTcpServer in pymodbus/server/async_io.py was originally accepting a **kwargs value in __init__ which is removed in current dev branch. If I'm right I think **kwargs should be added back, or the "loop" args in http_server.py L152 `server["loop"] = asyncio.get_running_loop()` should be removed.

### PR DESCRIPTION
**Pull request by fhydralisk**
Add a **kwargs init argument in ModbusTCPServer.

Also with unit test code added for this.

fixes #1662

### CODE REVIEW

File: pymodbus/server/async_io.py [hunk 1]
```diff
@@ -269,6 +269,7 @@ def __init__(
         broadcast_enable=False,
         response_manipulator=None,
         request_tracer=None,
+        **kwargs,  # pylint: disable=unused-argument
```
Comment (janiversen) [inline pymodbus/server/async_io.py:272]: This is wrong, it just hides the problem, it allows you to think that loop is an acceptable parameter.

Please instead remove the loop parameter in the call.

File: test/test_simulator.py [hunk 2]
```diff
@@ -539,3 +560,18 @@ def test_simulator_action_random(self, celltype, minval, maxval):
                     regs, is_int
                 )
             assert minval <= new_value <= maxval
+
```
Comment (janiversen) [inline test/test_simulator.py:563]: Nice test.

**Review by janiversen — CHANGES_REQUESTED** [review]

File: pymodbus/server/async_io.py (hunk 1 above)
Comment (fhydralisk) [reply pymodbus/server/async_io.py:272]: I found that in async_io.py ModbusSerialServer takes **kwargs while other types(Tcp, Tls,...) does not take this argument, so I think simply remove "loop" in the call may cause problems. Should I check server class before removing it?

Comment (janiversen) [reply pymodbus/server/async_io.py:272]: No the server classes get the loop internally with asyncio.get_current_loop(), so it no longer have an effect to add loop=

Kmarts still exist in some server classes due  to other parameters.

Comment (fhydralisk) [reply pymodbus/server/async_io.py]: Changes commited.

**Review by janiversen — APPROVED** [review]
LGTM, Thanks.

Looking forward to see more PRs from you.

**Pull request merged**